scipy/optimize/_linprog_util.py

```diff
@@ -147,80 +203,55 @@ def _clean_inputs(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None, bounds=None,
-    except TypeError:
-        raise TypeError(
-            "Invalid input for linprog: c must be a 1D array of numerical "
-            "coefficients")
 
+    sparse_lhs = sps.issparse(A_eq) or sps.issparse(A_ub)
     try:
-        try:
-            if sps.issparse(A_eq) or sps.issparse(A_ub):
-                A_ub = sps.coo_matrix(
-                    (0, n_x), dtype=float) if A_ub is None else sps.coo_matrix(
-                    A_ub, dtype=float).copy()
-            else:
-                A_ub = np.zeros(
-                    (0, n_x), dtype=float) if A_ub is None else np.asarray(
-                    A_ub, dtype=float).copy()
-        except BaseException:
-            raise TypeError
+        A_ub = _format_A_constraints(A_ub, n_x, sparse_lhs=sparse_lhs)
+    except ValueError:
+        raise TypeError(
+            'Invalid value for linprog: A_ub must be a 2D array_like '
+            'object of numerical values.')
+    else:
         n_ub = A_ub.shape[0]
-        if len(A_ub.shape) != 2 or A_ub.shape[1] != len(c):
+        if len(A_ub.shape) != 2 or A_ub.shape[1] != n_x:
             raise ValueError(
                 "Invalid input for linprog: A_ub must have exactly two "
                 "dimensions, and the number of columns in A_ub must be "
-                "equal to the size of c ")
+                "equal to the size of c.")
         if (sps.issparse(A_ub) and not np.isfinite(A_ub.data).all()
                 or not sps.issparse(A_ub) and not np.isfinite(A_ub).all()):
             raise ValueError(
                 "Invalid input for linprog: A_ub must not contain values "
                 "inf, nan, or None")
-    except TypeError:
-        raise TypeError(
-            "Invalid input for linprog: A_ub must be a numerical 2D array "
-            "with each row representing an upper bound inequality constraint")
 
     try:
-        try:
-            b_ub = np.array(
-                [], dtype=float) if b_ub is None else np.asarray(
-                b_ub, dtype=float).copy().squeeze()
-        except BaseException:
-            raise TypeError
-        if b_ub.size == 1:
-            b_ub = b_ub.reshape((-1))
-        if len(b_ub.shape) != 1:
+        b_ub = _format_b_constraints(b_ub)
+    except ValueError:
+        raise TypeError(
+            "Invalid input for linprog: b_ub must be a 1D array of "
+            "numerical values, each representing the upper bound of an "
+            "inequality constraint (row) in A_ub")
+    else:
+        if b_ub.shape != (n_ub,):
             raise ValueError(
                 "Invalid input for linprog: b_ub should be a 1D array; it "
-                "must not have more than one non-singleton dimension")
-        if len(b_ub) != n_ub:
-            raise ValueError(
-                "Invalid input for linprog: The number of rows in A_ub must "
-                "be equal to the number of values in b_ub")
+                "must not have more than one non-singleton dimension where "
```

Comment:
"Invalid input for linprog: b_ub should be a 1D array, and the number of values must equal the number of rows in A_ub."

Reply:
In any case, make it consistent with the message for b_eq

Reply:
done.